File: Problems/PerfectNumber/js/perfectnumber.js
/**
 * @param {number} num
 * @return {boolean}
 */
var checkPerfectNumber = function(num) {
    let i = 2, sum = 1;
    while (i * i <= num) {
        if (num % i === 0) {
            sum += i + num/i;
        }
        i += 1;
    }
    
    return sum === num && num !== 1 ? true: false;
};

File: Problems/PerfectNumber/py/perfectnumber.py
class Solution:        
    def checkPerfectNumber(self, num: int) -> bool:
        sum = 1
        i = 2
        while i*i <= num:
            if num % i == 0:
                sum = sum + i + num/i
            i += 1
            
        if sum == num and num != 1: return True
        else: return False